force-app/main/default/classes/SOQL.cls

```diff
@@ -1,361 +1,585 @@
-        return null; // Replace null with the variable you used to store the result
+    keyword = '%' + keyword + '%';
+
+    return [
+      SELECT Id, Name
+      FROM Opportunity
+      WHERE Name LIKE :keyword
+      WITH USER_MODE
+    ];
+  }
+
+  /**
+   * Question 9
+   * Retrieves all Cases that Case Origin is not Web or Email or Blank
+   * @return A list of Case records.
+   */
+  public static List<Case> getFilterCaseOrigins() {
+    return [
+      SELECT Id, CaseNumber, Origin
+      FROM Case
+      WHERE Origin != NULL AND Origin != '' AND (NOT Origin IN ('Email', 'Web'))
+    ];
+  }
+
+  /**
+   * Question 10
+   * Update/Uncomment the template code to create a method
+   * Retrieves all Leads that have a specific Lead Source or contain a specific keyword in their Title.
+   * Example: yourMethodName('Web', 'Engineer') should return all Leads that have 'Web' in their Lead Source or 'Engineer' in their Title.
+   * @param leadSource The Lead Source to search for.
+   * @param keyword The keyword to search for in the Lead Title.
+   * @return A list of Lead records.
+   * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
+   */
+  //[public | private ] [static] data_type method_name (input parameters) {
+  public static List<Lead> getLeadsBySourceOrTitle(
+    String leadSource,
+    String keyword
+  ) {
+    if (String.isBlank(leadSource) || String.isBlank(keyword)) {
+      return null;
     }
-    
-    /**
-     * Question 15
-     * Retrieves the Contacts with the most recent Birthdate, ordered by Birthdate, and limited to 3 records.
-     * @return A list of Contacts
-     */
-    public static List<Contact> getBirthdateContacts() {
-        return null; // Replace null with the variable you used to store the result
+    leadSource = '%' + leadSource + '%';
+    keyword = '%' + keyword + '%';
+    return [
+      SELECT Id, LastName, Title, LeadSource, Company, Phone, Email
+      FROM Lead
+      WHERE LeadSource LIKE :leadSource OR Title LIKE :keyword
+    ];
+  }
+
+  /**
+   * Question 11
+   * Returns the names of all Contacts associated with an Account.
+   * Example: getContactNamesByAccount('001XXXXXXXXXXXX') should return the list of Contact LastNames for the Account
+   *  with the Id
+   * @param accountId The ID of the Account.
+   * @return A list of Contact LastName.
+   */
+  public static List<String> getContactNamesByAccount(Id accountId) {
+    if (String.isBlank(accountId)) {
+      return null;
     }
-    
-    /**
-     * Question 16
-     * Retrieves the Cases in the Open Status, ordered by CreatedDate, and limited to 10 records with an offset of 5.
-     * @return A list of Cases
-     */
-    public static List<Case> getCasesWithStatus() {
-        return null; // Replace null with the variable you used to store the result
+    // Create a list to hold the Contact LastNames
+    List<String> ctctLastNameLst = new List<String>();
```

Comment:
I'm not ctctLastNameLst variable names.
I'd go with contactsLastNameList
It's okay to be more verbose.